internal/app/machined/pkg/system/runner/containerd/containerd.go

```diff
@@ -115,8 +116,17 @@ func (c *containerdRunner) Close() error {
 func (c *containerdRunner) Run(eventSink events.Recorder) error {
 	defer close(c.stopped)
 
+	var (
+		task containerd.Task
+		err  error
+	)
+
 	// Create the task and start it.
-	task, err := c.container.NewTask(c.ctx, cio.LogFile(c.logPath()))
+	if c.debug {
+		task, err = c.container.NewTask(c.ctx, cio.NewCreator(cio.WithStreams(os.Stdin, os.Stdout, os.Stderr)))
+	} else {
+		task, err = c.container.NewTask(c.ctx, cio.LogFile(c.logPath()))
+	}
```

Comment:
This could be simplified a bit:

```
	var (
		task containerd.Task
		err  error
	)

	creator := cio.LogFile(c.logPath())
	if c.debug {
		creator = cio.NewCreator(cio.WithStreams(os.Stdin, os.Stdout, os.Stderr))
	}

	// Create the task and start it.
	task, err = c.container.NewTask(c.ctx, creator)
	if err != nil {
		return fmt.Errorf("failed to create task: %q: %w", c.args.ID, err)
	}
```